### PR TITLE
[QC-15] [QC-293] [QC-252] Multinode setups

### PR DESCRIPTION
**Pull request by knopers8**
That works with my branch in O2 https://github.com/knopers8/AliceO2/tree/rework-mergers
TODO:

- [x] Finish reworking Mergers and merge them to O2
- [x] Make sure detector modules are properly updated where needed (little changes in config files + use `generateStandaloneInfrastracture` in their development executables)
- [x] Clang-format
- [x] Update tests accordingly
- [x] Documentation
- [x] Nice new example. Make runAdvanced still advanced, but do not try to simulate a multinode setups there anymore.
- [x] Support multinode setups with Data Sampling - a chain with local Dispatcher and remote QC task (I might do that later though, in another PR)

### CODE REVIEW

**Comment by knopers8** [comment]
@JianLIUhep If you have time, could you give it a try?

**Comment by JianLIUhep** [comment]
@knopers8 Thanks a lot for the notice. Yes, we will test this new version with our ITS setup asap.

File: doc/Advanced.md [hunk 1]
```diff
@@ -40,6 +40,121 @@ For example, if TPC wants to monitor the output `{"TPC", "CLUSTERS"}` of the wor
...
+## Multi-node setups
+
+During the data-taking Quality Control runs on a distributed computing system. Some QC Tasks are
+executed on dedicated QC servers, while others run on FLPs and EPNs. In the first case, messages
+coming from Data Sampling should reach QC servers where they are processed. In the latter case,
+locally produced Monitor Objects should be merged on QC servers and then have Checks run on them.
+
+While it is responsibility of the run operators to run all the processing topologies during the
+data taking, here we show how to achieve such multinode workflows on development setups, running
+them just with DPL driver. Note that for now we support cases with one or more local machines,
```
Comment (Barthelemy) [inline doc/Advanced.md:52]: clarify what "local" means in this context.

File: doc/Advanced.md [hunk 2]
```diff
@@ -40,6 +40,121 @@ For example, if TPC wants to monitor the output `{"TPC", "CLUSTERS"}` of the wor
...
+o2-qc-run-producer | o2-qc --config json:/${QUALITYCONTROL_ROOT}/etc/multiNode.json --local --host localnode1 -b
+# On localnode2:
+o2-qc-run-producer | o2-qc --config json:/${QUALITYCONTROL_ROOT}/etc/multiNode.json --local --host localnode2 -b
+# On qcnode:
+o2-qc --config json:/${QUALITYCONTROL_ROOT}/etc/multiNode.json --remote
+```
+
+If there are no problems, on QCG you should see the `example` histogram updated under the paths `qc/TST/MultiNodeLocal`
+and `qc/TST/MultiNodeRemote`, and corresponding Checks under the path `qc/checks/TST/`.
+
```
Comment (Barthelemy) [inline doc/Advanced.md:157]: very nice doc

**Comment by knopers8** [comment]
Thanks for the review, I was fixing the test as were reviewing, it should be green now.

**Comment by knopers8** [comment]
Please put `Multi-node QC setups` as a commit title when squashing

**Comment by JianLIUhep** [comment]
@knopers8 We tested this feature on 3 ITS FLPs (2 local + 1 "remote") and it seems to be working fine with the Skeleton task! We will try to make our ITS task work asap. One comment on the firewall setting: if one does not want to use general iptables, one needs to open the "remote machine port", "data sampling ports" and **"port 7777"**.  It did not work if port 7777 not opened.

**Comment by knopers8** [comment]
@JianLIUhep Thanks for checking it out, this is good news that it works!
I had no idea about the port 7777, do you know what is it used for? How did you find out?

**Comment by JianLIUhep** [comment]
@knopers8 We initially just opened the ports for the remote machine and data sampling, but it did not work. Then, we temporarily switched off the firewall and saw the traffic through port 7777.  For my understanding, "QC-TASK-RUNNER-MultiNodeRemote" occupies this port taking this default value of 7777 somewhere.

**Comment by knopers8** [comment]
Thank you, i will investigate that

**Pull request merged**